### PR TITLE
CMCL-0000: lens property drawer refactor

### PR DESCRIPTION
**Pull request by glabute**
### Purpose of this PR

Th code was overly-complex.  This PR simplifies it and makes it more maintainable.

### Testing status

- [ ] Added an automated test
- [ ] Passed all automated tests
- [x] Manually tested

### CODE REVIEW

File: com.unity.cinemachine/Editor/PropertyDrawers/LensSettingsPropertyDrawer.cs [hunk 1]
```diff
@@ -53,35 +53,33 @@ System.Reflection.BindingFlags bindingFlags
...
-                return default(T);
+                return default;
         }
 
         static bool s_PhysicalExapnded;
 
         static List<string> m_PresetOptions;
         static List<string> m_PhysicalPresetOptions;
-        static string m_EditPresetsLabel = "Edit Presets...";
+        const string kEditPresetsLabel = "Edit Presets...";
```
Comment (gaborkb) [inline com.unity.cinemachine/Editor/PropertyDrawers/LensSettingsPropertyDrawer.cs:63]: k_

**Review by gaborkb — CHANGES_REQUESTED** [review]
I get the following error in ThirdPerson Shooter sample scene, when I click on StateDrivenCamera:
```
NullReferenceException: Object reference not set to an instance of an object
UnityEditor.EditorGUIUtility.TempContent (System.String[] texts) (at /Users/bokken/build/output/unity/unity/Editor/Mono/EditorGUIUtility.cs:1062)
UnityEditor.EditorGUI.Popup (UnityEngine.Rect position, System.Int32 selectedIndex, System.String[] displayedOptions, UnityEngine.GUIStyle style) (at /Users/bokken/build/output/unity/unity/Editor/Mono/EditorGUI.cs:8544)
UnityEditor.EditorGUI.Popup (UnityEngine.Rect position, System.Int32 selectedIndex, System.String[] displayedOptions) (at /Users/bokken/build/output/unity/unity/Editor/Mono/EditorGUI.cs:8539)
Cinemachine.Editor.CinemachineStateDrivenCameraEditor+<>c__DisplayClass17_0.<SetupInstructionList>b__2 (UnityEngine.Rect rect, System.Int32 index, System.Boolean isActive, System.Boolean isFocused) (at /Users/gaborkb/Documents/Dev/Packages/cinemachine/upm-package-cinemachine/com.unity.cinemachine/Editor/Editors/CinemachineStateDrivenCameraEditor.cs:289)
UnityEditorInternal.ReorderableList.DoListElements (UnityEngine.Rect listRect, UnityEngine.Rect visibleRect) (at /Users/bokken/build/output/unity/unity/Editor/Mono/GUI/ReorderableList.cs:985)
UnityEditorInternal.ReorderableList.DoLayoutList () (at /Users/bokken/build/output/unity/unity/Editor/Mono/GUI/ReorderableList.cs:750)
Cinemachine.Editor.CinemachineStateDrivenCameraEditor+<>c__DisplayClass10_0.<CreateInspectorGUI>b__7 () (at /Users/gaborkb/Documents/Dev/Packages/cinemachine/upm-package-cinemachine/com.unity.cinemachine/Editor/Editors/CinemachineStateDrivenCameraEditor.cs:72)
UnityEngine.UIElements.IMGUIContainer.DoOnGUI (UnityEngine.Event evt, UnityEngine.Matrix4x4 parentTransform, UnityEngine.Rect clippingRect, System.Boolean isComputingLayout, UnityEngine.Rect layoutSize, System.Action onGUIHandler, System.Boolean canAffectFocus) (at /Users/bokken/build/output/unity/unity/ModuleOverrides/com.unity.ui/Core/IMGUIContainer.cs:381)
UnityEditor.InspectorWindow:RedrawFromNative() (at /Users/bokken/build/output/unity/unity/Editor/Mono/Inspector/InspectorWindow.cs:201)

```

File: com.unity.cinemachine/Editor/PropertyDrawers/LensSettingsPropertyDrawer.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ class LensSettingsHideModeOverridePropertyDrawer : LensSettingsPropertyDrawer
     [CustomPropertyDrawer(typeof(LensSettings))]
     class LensSettingsPropertyDrawer : PropertyDrawer
     {
-        static LensSettings m_LensSettingsDef = new LensSettings(); // to access name strings
+        static LensSettings m_LensSettingsDef = new (); // to access name strings
```
Comment (gaborkb) [inline com.unity.cinemachine/Editor/PropertyDrawers/LensSettingsPropertyDrawer.cs:25]: s_
for static I think

File: com.unity.cinemachine/Editor/PropertyDrawers/LensSettingsPropertyDrawer.cs [hunk 3]
```diff
@@ -191,309 +182,257 @@ void DoUpdate()
...
                     }
-                    // Push to property
-                    if (!Mathf.Approximately(focalProperty.floatValue, vfov))
+                    case Modes.Physical:
                     {
-                        focalProperty.floatValue = vfov;
-                        focalProperty.serializedObject.ApplyModifiedProperties();
+                        // Convert to display FolcalLength units
+                        var fovProp = m_LensProperty.FindPropertyRelative(() => m_LensSettingsDef.FieldOfView);
+                        var value = Camera.FieldOfViewToFocalLength(fovProp.floatValue, SensorSize(m_LensProperty).y);
```
Comment (gaborkb) [inline com.unity.cinemachine/Editor/PropertyDrawers/LensSettingsPropertyDrawer.cs:288]: `var value` hides inherited property `value`. Inherited from BaseField.

File: com.unity.cinemachine/Editor/PropertyDrawers/LensSettingsPropertyDrawer.cs [hunk 4]
```diff
@@ -191,309 +182,257 @@ void DoUpdate()
...
-                    else 
+                    case Modes.VFOV:
+                    case Modes.HFOV:
                     {
-                        // Apply the preset
-                        var index = CinemachineLensPresets.Instance.GetPhysicalPresetIndex(evt.newValue);
-                        if (index >= 0)
+                        // Convert to display FOV units
+                        var fovProp = m_LensProperty.FindPropertyRelative(() => m_LensSettingsDef.FieldOfView);
+                        var value = fovProp.floatValue;
```
Comment (gaborkb) [inline com.unity.cinemachine/Editor/PropertyDrawers/LensSettingsPropertyDrawer.cs:298]: `var value` hides inherited property `value`. Inherited from BaseField.

File: com.unity.cinemachine/Editor/PropertyDrawers/LensSettingsPropertyDrawer.cs [hunk 5]
```diff
@@ -191,309 +182,257 @@ void DoUpdate()
...
+                            fovProp.serializedObject.ApplyModifiedProperties();
+                        }
                     }
                 }
-                else
+            }
+
+            void SyncPresetDisplay(float lensValue, bool physical)
+            {
+                var value = string.Empty;
```
Comment (gaborkb) [inline com.unity.cinemachine/Editor/PropertyDrawers/LensSettingsPropertyDrawer.cs:418]: var value hides inherited property value. Inherited from BaseField.

**Review by gaborkb — CHANGES_REQUESTED** [review]

**Comment by glabute** [comment]
I get the exception on main branch also, it's not related to this PR

**Review by gaborkb — APPROVED** [review]

**Pull request merged**